src/Build/BackEnd/Components/Communications/NodeProviderOutOfProc.cs

```diff
@@ -72,17 +72,22 @@ public int AvailableNodes
         /// Derived from the binary timestamp to avoid mixing binary versions,
         /// Is64BitProcess to avoid mixing bitness, and enableNodeReuse to
         /// ensure that a /nr:false build doesn't reuse clients left over from
-        /// a prior /nr:true build.
+        /// a prior /nr:true build. The enableLowPriority flag is to ensure that
+        /// a build with /low:false doesn't reuse clients left over for a prior
+        /// /low:true build.
         /// </summary>
         /// <param name="enableNodeReuse">Is reuse of build nodes allowed?</param>
-        internal static long GetHostHandshake(bool enableNodeReuse)
+        /// <param name="enableLowPriority">Is the build running at low priority?</param>
+        internal static long GetHostHandshake(bool enableNodeReuse, bool enableLowPriority)
         {
             long baseHandshake = Constants.AssemblyTimestamp;
 
             baseHandshake = baseHandshake*17 + EnvironmentUtilities.Is64BitProcess.GetHashCode();
 
             baseHandshake = baseHandshake*17 + enableNodeReuse.GetHashCode();
 
+            baseHandshake = baseHandshake*17 + enableLowPriority.GetHashCode();
```

Comment:
BTW if you do need more bits (eg., to specify the fine grained priority) you can probably remove the masking of the first byte that is done in `GetClientHandshake()`. This masking was needed when nodes might exist running versions of MSBuild written before ~2008 (?) ... which I think is now unlikely.

Reply:
As you pointed out below, there aren't specific bits for features anymore, but I could theoretically take advantage of this masking to make the first byte the priority. (I could really probably use the first four bits.) That would be a bit complicated, though, so I'm not sure it's worth it.

Reply:
Both sides of the handshake should be the same exact binaries, so this could theoretically be changed to be bit flipping. Not suggesting you do that, but it's not as if this scheme is locked in.

(Well - except that if this scheme is making a truly universally distributed bit pattern - then it actually is impossible to be certain that you won't accidentally connect to it. Unless you make the handshake longer. But now we're being even more implausible.)